main.py

```diff
@@ -113,10 +120,25 @@ async def get_compiled_artifact(task_id: str) -> Dict:
     return results[task_id].dict()
 
 
-async def compile_project(project_root: Path, files: List[UploadFile]):
+async def compile_project(project_root: Path, config_override: dict):
     """
     Compile the contrct and asssign the taskid to it
     """
+    (project_root/"ape-config.yaml").write_text(yaml.safe_dump(config_override))
+
     with config.using_project(project_root) as project:
-        results[project_root.name] = project.extract_manifest()
-    tasks[project_root.name] = TaskStatus.SUCCESS
+        # TODO Handle and add more Error Types to except 
+        
+        # compile contracts in folder wholesale
+        try:
+            project.load_contracts(use_cache=False)
+            results[project_root.name] = project.extract_manifest()
+            tasks[project_root.name] = TaskStatus.SUCCESS
+        except VyperInstallError as e:
```

Comment:
**todo**: you can except multiple exceptions in one line when their handling is the same

```python
except (VyperInstallError, VyperCompileError):
    results[project_root.name] = [str(e)]
    tasks[project_root.name] = TaskStatus.FAILED
```

Reply:
we want to be explicit with the errors. if it get too much, ill change it

Reply:
Hmm, maybe this isn't the right path to go

Actually, what we can do is not make this rely on the vyper plugin at all by using `ape.exceptions.CompileError`